### DISCUSSION TITLE
Local imports not working in api functions

### DISCUSSION THREAD

**Discussion by MarcGuiselin** [nuxt/nuxt#18200] [Questions]
### Edit: This might be a bug

This is probably a bug, because `import message from 'test-lib/test'` works fine in `/app.vue` but `import message from 'test-lib/test'` doesn't work in `/server/api/test.ts`.

I've updated the repo to demonstrate this. There just seem to be contexts (like the api) where importing non-js like typescript doesn't work.

### Original

As far as I understand, with Nuxt 3 it isn't possible to import typescript/vue/non-js files from another package within a monorepo without an additional compile step.

The only solution right now is compiling local packages to an esm first. Once you do this, nuxt can import it no problem. Having to recompile local packages manually after changing them and restarting nuxt isn't practical.

I'm looking for a better solution.

### Minimum Example

I have a test monorepo configured with Nuxt and a test library called `test-lib`, both using typescript: https://github.com/MarcGuiselin/nuxt3-monorepo-import-bug

From Nuxt, if I try to import a file from the test library, I get the following error:
`WARN  [worker] Cannot find module '/home/marc/nuxt3-monorepo-import-bug/node_modules/test-lib/dir/test' imported from /home/marc/nuxt3-monorepo-import-bug/packages/nuxt3-app/.nuxt/nitro/index.mjs`

The error is confusing but expected since `test-lib` is typescript and hasn't been compiled into a js module.

### Solution for Nextjs

In next.js, this issue would be solved using the [`next-transpile-modules`](https://github.com/martpie/next-transpile-modules) npm package. Transpiling the package `test-lib` would allow me to import and use it throughout my app.

Internally `next-transpile-modules` works simply by altering a few webpack rules during compilation. I'm unsure how to implement this with Nuxt, since [extending the webpack config](https://nuxtjs.org/docs/features/configuration/#extend-webpack-config) isn't possible anymore.

### Nuxt's build.transpile is not a solution

In Nuxt (as far as I understand) [`build.transpile`](https://nuxtjs.org/docs/configuration-glossary/configuration-build/#transpile) can be configured to fix this. However, setting `build.transpile = ['test-lib']` doesn't work.

In fact, using the following `nuxt.config.ts` never logs the `test-lib` library, so it seems that `test-lib` is being omitted from the transpile step. (I might be completely misusing transpile here, I'm not sure)
````ts
import { defineNuxtConfig } from 'nuxt3'

export default defineNuxtConfig({
  build: {
    transpile: [
      (filename: string) => {
        console.log(filename)
      },
    ],
  },
})
````

**Comment by peterbud** [comment]
Was there  any solution for this? It seems to me that this is still an issue
